testing/test_assertrewrite.py

```diff
@@ -1116,8 +1120,15 @@ def test_read_pyc_success(self, tmp_path: Path, pytester: Pytester) -> None:
 
         fn.write_text("def test(): assert True", encoding="utf-8")
 
-        source_stat, co = _rewrite_test(fn, config)
-        _write_pyc(state, co, source_stat, pyc)
+        source_stat, hash, co = _rewrite_test(fn, config)
+        _write_pyc(state, co, source_stat, hash, pyc)
+        assert _read_pyc(fn, pyc, state.trace) is not None
+
+        # pyc read should still work if only the mtime changed
+        # Fallback to hash comparison
+        sleep(0.1)
```

Comment:
`sleep` slows down the testsuite. For this use `os.utime` to set the mtime instead?